In [3]:
import os
import cv2
import statistics

In [1]:
path = "../all_videos"
dataset_path = "../gte9_dataset"

num of vds for every worf

In [6]:
freq={}
# {freq = [numofword,[wordlist]]}

vids_done = 0

for i in range(17):
    freq[i] = [0,[]]
for word_folder in os.listdir(dataset_path):
    
    numofvids = len(os.listdir(os.path.join(dataset_path,word_folder)))
    
    
    freq[numofvids][0]+=1
    freq[numofvids][1].append(word_folder)

        

print(freq)

# import json

# with open("stats_freq.json","w") as f:
#     json.dump(freq,f)

             
            



{0: [0, []], 1: [0, []], 2: [0, []], 3: [0, []], 4: [0, []], 5: [0, []], 6: [0, []], 7: [0, []], 8: [0, []], 9: [100, ['add', 'africa', 'ago', 'alone', 'animal', 'australia', 'baby', 'ball', 'balloon', 'banana', 'barely', 'beard', 'big', 'buy', 'california', 'can', 'careful', 'carrot', 'cat', 'catch', 'chat', 'child', 'city', 'class', 'correct', 'country', 'cow', 'crash', 'day', 'decide', 'decorate', 'dirty', 'dive', 'drop', 'dry', 'fast', 'fault', 'fine', 'finish', 'first', 'follow', 'football', 'form', 'future', 'girl', 'glasses', 'grammar', 'great', 'hair', 'halloween', 'hat', 'headache', 'hear', 'hope', 'how', 'improve', 'inform', 'kiss', 'list', 'lose', 'make', 'meet', 'minute', 'near', 'necklace', 'new', 'now', 'office', 'outside', 'party', 'past', 'pink', 'point', 'practice', 'president', 'problem', 'pull', 'right', 'same', 'school', 'shock', 'show', 'sign', 'stay', 'tiger', 'today', 'town', 'train', 'travel', 'ugly', 'upset', 'visit', 'voice', 'vomit', 'want', 'water', 'week', 

In [4]:
lessthan = {"lessthan 0":0}
for i in range(1,17):
    lessthan[f"lessthan {i}"] = lessthan[f"lessthan {i-1}"] + freq[i-1][0]
    
    
print(lessthan)


{'lessthan 0': 0, 'lessthan 1': 0, 'lessthan 2': 0, 'lessthan 3': 14, 'lessthan 4': 90, 'lessthan 5': 425, 'lessthan 6': 936, 'lessthan 7': 1338, 'lessthan 8': 1645, 'lessthan 9': 1796, 'lessthan 10': 1896, 'lessthan 11': 1944, 'lessthan 12': 1970, 'lessthan 13': 1983, 'lessthan 14': 1991, 'lessthan 15': 1995, 'lessthan 16': 1997}


list of less than5:

In [5]:
lessthan5 = []

for i in range(5):
    lessthan5.extend(freq[i][1])

lessthan5.sort()
print(len(lessthan5))
print(lessthan5)

425
['a', 'address', 'adjective', 'adjust', 'advanced', 'advantage', 'adverb', 'agenda', 'ahead', 'alcohol', 'alphabet', 'anatomy', 'annoy', 'anyway', 'arizona', 'article', 'asl', 'assume', 'attorney', 'audiologist', 'audiology', 'austria', 'author', 'available', 'award', 'aware', 'b', 'babysitter', 'bank', 'basic', 'battery', 'battle', 'bible', 'bike', 'binoculars', 'boast', 'boil', 'bone', 'bookshelf', 'bookstore', 'boots', 'boxing', 'bra', 'braid', 'brain', 'breakdown', 'breeze', 'bribe', 'bridge', 'brochure', 'bug', 'burp', 'cabbage', 'camel', 'canoe', 'captain', 'cards', 'careless', 'caterpillar', 'cheerleader', 'chicago', 'china', 'choke', 'christ', 'click', 'collect', 'comma', 'commit', 'company', 'compete', 'concentrate', 'concept', 'congratulations', 'conquer', 'consider', 'consume', 'contest', 'contract', 'convert', 'cooperate', 'counselor', 'court', 'create', 'crocodile', 'crown', 'cruel', 'crush', 'cuba', 'culture', 'curtain', 'customer', 'd', 'damage', 'dancer', 'dawn', 'd

greater than equal to 9

In [ ]:
gte9 = []

for i in range(9,17):
    gte9.extend(freq[i][1])

gte9.sort()
print(len(gte9))
print(gte9)



204
['accident', 'add', 'africa', 'ago', 'alone', 'animal', 'apple', 'appointment', 'argue', 'australia', 'baby', 'bad', 'balance', 'ball', 'balloon', 'banana', 'bar', 'barely', 'basketball', 'beard', 'because', 'bed', 'before', 'big', 'bird', 'black', 'blanket', 'bowling', 'brother', 'buy', 'california', 'call', 'can', 'candy', 'careful', 'carrot', 'cat', 'catch', 'champion', 'change', 'chat', 'cheat', 'check', 'child', 'city', 'class', 'cold', 'computer', 'convince', 'cool', 'corn', 'correct', 'country', 'cousin', 'cow', 'crash', 'cry', 'dark', 'daughter', 'day', 'deaf', 'decide', 'decorate', 'delay', 'delicious', 'dirty', 'dive', 'doctor', 'dog', 'drink', 'drop', 'dry', 'environment', 'example', 'family', 'far', 'fast', 'fat', 'fault', 'fine', 'finish', 'first', 'fish', 'follow', 'football', 'form', 'full', 'future', 'girl', 'give', 'glasses', 'go', 'good', 'government', 'graduate', 'grammar', 'great', 'hair', 'halloween', 'hat', 'headache', 'hear', 'help', 'hope', 'hot', 'how', 'im

In [9]:
print(gte9)
gte9_newline = [x+"\n" for x in gte9]
print(gte9)
with open("gte9_list.txt","w") as f5:
    f5.writelines(gte9_newline)

['accident', 'add', 'africa', 'ago', 'alone', 'animal', 'apple', 'appointment', 'argue', 'australia', 'baby', 'bad', 'balance', 'ball', 'balloon', 'banana', 'bar', 'barely', 'basketball', 'beard', 'because', 'bed', 'before', 'big', 'bird', 'black', 'blanket', 'bowling', 'brother', 'buy', 'california', 'call', 'can', 'candy', 'careful', 'carrot', 'cat', 'catch', 'champion', 'change', 'chat', 'cheat', 'check', 'child', 'city', 'class', 'cold', 'computer', 'convince', 'cool', 'corn', 'correct', 'country', 'cousin', 'cow', 'crash', 'cry', 'dark', 'daughter', 'day', 'deaf', 'decide', 'decorate', 'delay', 'delicious', 'dirty', 'dive', 'doctor', 'dog', 'drink', 'drop', 'dry', 'environment', 'example', 'family', 'far', 'fast', 'fat', 'fault', 'fine', 'finish', 'first', 'fish', 'follow', 'football', 'form', 'full', 'future', 'girl', 'give', 'glasses', 'go', 'good', 'government', 'graduate', 'grammar', 'great', 'hair', 'halloween', 'hat', 'headache', 'hear', 'help', 'hope', 'hot', 'how', 'improv

In [ ]:
print(lessthan5)
lessthan5_newline = [x+"\n" for x in lessthan5]
print(lessthan5_newline)



['a', 'address', 'adjective', 'adjust', 'advanced', 'advantage', 'adverb', 'agenda', 'ahead', 'alcohol', 'alphabet', 'anatomy', 'annoy', 'anyway', 'arizona', 'article', 'asl', 'assume', 'attorney', 'audiologist', 'audiology', 'austria', 'author', 'available', 'award', 'aware', 'b', 'babysitter', 'bank', 'basic', 'battery', 'battle', 'bible', 'bike', 'binoculars', 'boast', 'boil', 'bone', 'bookshelf', 'bookstore', 'boots', 'boxing', 'bra', 'braid', 'brain', 'breakdown', 'breeze', 'bribe', 'bridge', 'brochure', 'bug', 'burp', 'cabbage', 'camel', 'canoe', 'captain', 'cards', 'careless', 'caterpillar', 'cheerleader', 'chicago', 'china', 'choke', 'christ', 'click', 'collect', 'comma', 'commit', 'company', 'compete', 'concentrate', 'concept', 'congratulations', 'conquer', 'consider', 'consume', 'contest', 'contract', 'convert', 'cooperate', 'counselor', 'court', 'create', 'crocodile', 'crown', 'cruel', 'crush', 'cuba', 'culture', 'curtain', 'customer', 'd', 'damage', 'dancer', 'dawn', 'debat

In [7]:
with open("lessthan5list.txt","w") as f5:
    f5.writelines(lessthan5_newline)

----------
num of videos for every word

In [5]:
freq={}
# {freq = [numofword,[wordlist]]}

vids_done = 0


for filename in os.listdir(path):
        if filename.lower().endswith('.mp4'):
            
            vid_path = os.path.join(path, filename)
            

            cap = cv2.VideoCapture(vid_path)
            
            fps = cap.get(cv2.CAP_PROP_FPS)
            fps_list.append(fps)
            
            
            frames = cap.get(cv2.CAP_PROP_FRAME_COUNT)
            frames_list.append(frames)
            
            
            
            cap.release()
            
            vids_done +=1
            print(vids_done)
            
            
print(len(fps_list))    
print(len(frames_list))
print("FPS")
getallstats(fps_list)
print("\n\n")

print("FRAMES")
getallstats(frames_list)



NameError: name 'fps_list' is not defined

In [7]:
def getallstats(data):
    # Calculate the mean (average)
    mean_value = statistics.mean(data)
    print(f"Mean: {mean_value}")

    # Calculate the median (middle value)
    median_value = statistics.median(data)
    print(f"Median: {median_value}")

    # Calculate the mode (most common value)
    # Note: statistics.mode() raises a StatisticsError if there is no unique mode.
    try:
        mode_value = statistics.mode(data)
        print(f"Mode: {mode_value}")
    except statistics.StatisticsError as e:
        print(f"Mode error: {e}")

    # Calculate the standard deviation (sample)
    stdev_value = statistics.stdev(data)
    print(f"Standard Deviation (sample): {stdev_value}")

    # Calculate the population standard deviation
    pstdev_value = statistics.pstdev(data)
    print(f"Standard Deviation (population): {pstdev_value}")

    # Calculate the variance (sample)
    variance_value = statistics.variance(data)
    print(f"Variance (sample): {variance_value}")

    # Calculate the population variance
    pvariance_value = statistics.pvariance(data)
    print(f"Variance (population): {pvariance_value}")

    # Calculate the harmonic mean
    harmonic_mean_value = statistics.harmonic_mean(data)
    print(f"Harmonic Mean: {harmonic_mean_value}")

In [14]:
fps_list = []
frames_list = []

vids_done = 0


for word in os.listdir(dataset_path):
    wordpath = os.path.join(dataset_path,word)
    
    for filename in os.listdir(wordpath):
    
        if filename.lower().endswith('.mp4'):
            
            vid_path = os.path.join(wordpath, filename)
            
            print(vid_path)
            cap = cv2.VideoCapture(vid_path)
            
            fps = cap.get(cv2.CAP_PROP_FPS)
            fps_list.append(fps)
            
            
            frames = cap.get(cv2.CAP_PROP_FRAME_COUNT)
            frames_list.append(frames)
            
            
            
            cap.release()
            
            vids_done +=1
            print(vids_done)
            
            
print(len(fps_list))    
print(len(frames_list))
print("FPS")
getallstats(fps_list)
print("\n\n")

print("FRAMES")
getallstats(frames_list)



../gte9_dataset\accident\00623.mp4
1
../gte9_dataset\accident\00624.mp4
2
../gte9_dataset\accident\00625.mp4
3
../gte9_dataset\accident\00626.mp4
4
../gte9_dataset\accident\00627.mp4
5
../gte9_dataset\accident\00628.mp4
6
../gte9_dataset\accident\00629.mp4
7
../gte9_dataset\accident\00631.mp4
8
../gte9_dataset\accident\00632.mp4
9
../gte9_dataset\accident\00633.mp4
10
../gte9_dataset\accident\00634.mp4
11
../gte9_dataset\accident\00639.mp4
12
../gte9_dataset\accident\65009.mp4
13
../gte9_dataset\add\00961.mp4
14
../gte9_dataset\add\00962.mp4
15
../gte9_dataset\add\00963.mp4
16
../gte9_dataset\add\00964.mp4
17
../gte9_dataset\add\00965.mp4
18
../gte9_dataset\add\00967.mp4
19
../gte9_dataset\add\00973.mp4
20
../gte9_dataset\add\65018.mp4
21
../gte9_dataset\add\69205.mp4
22
../gte9_dataset\africa\01383.mp4
23
../gte9_dataset\africa\01384.mp4
24
../gte9_dataset\africa\01385.mp4
25
../gte9_dataset\africa\01386.mp4
26
../gte9_dataset\africa\01387.mp4
27
../gte9_dataset\africa\01388.mp4
28
..

In [15]:
print(len(fps_list))    
print(len(frames_list))
print("FPS")
getallstats(fps_list)
print("\n\n")

print("FRAMES")
getallstats(frames_list)
print(max(fps_list))
print(max(frames_list))
print()
print(min(fps_list))
print(min(frames_list))

2060
2060
FPS
Mean: 28.380253905162473
Median: 29.97
Mode: 29.97
Standard Deviation (sample): 2.8451461515022367
Standard Deviation (population): 2.8444554982206682
Variance (sample): 8.094856623407988
Variance (population): 8.09092708135779
Harmonic Mean: 28.077939769876345



FRAMES
Mean: 65.12864077669903
Median: 63.0
Mode: 34.0
Standard Deviation (sample): 25.926586965929044
Standard Deviation (population): 25.92029334113285
Variance (sample): 672.1879117018818
Variance (population): 671.8616068903762
Harmonic Mean: 55.00679800136238
50.0
195.0

12.0
19.0


In [11]:
print(len(fps_list))    
print(len(frames_list))
print("FPS")
getallstats(fps_list)
print("\n\n")

print("FRAMES")
getallstats(frames_list)

2060
2060
FPS
Mean: 0.0
Median: 0.0
Mode: 0.0
Standard Deviation (sample): 0.0
Standard Deviation (population): 0.0
Variance (sample): 0.0
Variance (population): 0.0
Harmonic Mean: 0



FRAMES
Mean: 0.0
Median: 0.0
Mode: 0.0
Standard Deviation (sample): 0.0
Standard Deviation (population): 0.0
Variance (sample): 0.0
Variance (population): 0.0
Harmonic Mean: 0


11980
11980
FPS
Mean: 28.530534785232394
Median: 29.97
Mode: 29.97
Standard Deviation (sample): 3.101906651411547
Standard Deviation (population): 3.1017771868295334
Variance (sample): 9.621824874071198
Variance (population): 9.621021716736132
Harmonic Mean: 28.20762657332378



FRAMES
Mean: 69.12295492487479
Median: 70.0
Mode: 85.0
Standard Deviation (sample): 26.64041389834537
Standard Deviation (population): 26.639302004778727
Variance (sample): 709.7116526751531
Variance (population): 709.652411301808
Harmonic Mean: 58.530410392450605

In [ ]:

minfps = 1e9
maxfps = -1e9
totalfps = 0

minframes = 1e9
maxframes = -1e9
totalframes = 0



vids_done = 0


for filename in os.listdir(path):
        if filename.lower().endswith('.mp4'):
            
            vid_path = os.path.join(path, filename)
            

            cap = cv2.VideoCapture(vid_path)
            fps = cap.get(cv2.CAP_PROP_FPS)
            minfps = min(minfps, fps)
            maxfps = max(maxfps,fps)
            totalfps += fps
            
            
            frames = cap.get(cv2.CAP_PROP_FRAME_COUNT)
            maxframes = max(maxframes,frames)
            minframes = min(maxframes,frames)
            totalframes += frames
            if frames > 80:
                print(vid_path)
            
            
            cap.release()
            
            vids_done +=1
            print(vids_done)
            
            
print(minfps)
print(maxfps)
print(totalfps/vids_done)
print(minframes)
print(maxframes)
print(totalframes/vids_done)


"""
12.0
59.94
28.530534785231108
57.0
233.0
69.12295492487479
"""



